bsuir_repo_back/users/views/profile.py

```diff
@@ -0,0 +1,60 @@
+from rest_framework import viewsets, status, permissions, parsers
+from rest_framework.response import Response
+from drf_yasg.utils import swagger_auto_schema
+
+from bsuir_repo_core.swagger_service import apply_swagger_auto_schema
+from users.models import UserProfile
+from users.permissions import IsBlocked
+from users.serializers.profile_serializer import ProfilePatchSerializer, ProfileSerializer
+
+
+class ProfileViewSet(viewsets.ModelViewSet):
+    parser_classes = [parsers.JSONParser, parsers.MultiPartParser]
+    permissions = [permissions.IsAuthenticated, IsBlocked]
+
+    def get_serializer_class(self):
+        if self.action == 'partial_update':
+            return ProfilePatchSerializer
+        return ProfileSerializer
+
+    def get_queryset(self):
+        user = self.request.user
+        if user.is_superuser:
+            users = UserProfile.objects.all()
```

Comment:
objects.all() make a lot n+1 ploblems, use select_related and other optimize methods